src/core/lib/event_engine/shim.cc

```diff
@@ -43,7 +39,7 @@ bool UsePollsetAlternative() {
 // Returns true if GRPC_DO_NOT_INSTANTIATE_POSIX_POLLER is defined.
 bool EventEngineExperimentDisabledForPython() {
 #ifdef GRPC_DO_NOT_INSTANTIATE_POSIX_POLLER
-  return true;
+  return !grpc_core::IsEventEnginePollerForPythonEnabled();
```

Comment:
nit: change the comment of this function. It doesn't always return true if GRPC_DO_NOT_INSTANTIATE_POSIX_POLLER is defined.